==66586== Memcheck, a memory error detector
==66586== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==66586== Using Valgrind-3.20.0 and LibVEX; rerun with -h for copyright info
==66586== Command: medusa -h 192.168.1.100 -u msfadmin -P password_list.txt -M ssh -n 22
==66586== Parent PID: 1333
==66586== 
==66586== 
==66586== HEAP SUMMARY:
==66586==     in use at exit: 8,448 bytes in 83 blocks
==66586==   total heap usage: 10,117 allocs, 10,034 frees, 1,913,488 bytes allocated
==66586== 
==66586== 9 bytes in 1 blocks are indirectly lost in loss record 1 of 19
==66586==    at 0x4843808: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x10F67A: loadLoginInfo (in /usr/bin/medusa)
==66586==    by 0x10C9CE: main (in /usr/bin/medusa)
==66586== 
==66586== 9 bytes in 1 blocks are definitely lost in loss record 2 of 19
==66586==    at 0x4843808: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x10E1EA: checkOptions (in /usr/bin/medusa)
==66586==    by 0x10C907: main (in /usr/bin/medusa)
==66586== 
==66586== 14 bytes in 1 blocks are indirectly lost in loss record 3 of 19
==66586==    at 0x4843808: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x10F770: loadLoginInfo (in /usr/bin/medusa)
==66586==    by 0x10C9CE: main (in /usr/bin/medusa)
==66586== 
==66586== 14 bytes in 1 blocks are definitely lost in loss record 4 of 19
==66586==    at 0x4843808: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x10E380: checkOptions (in /usr/bin/medusa)
==66586==    by 0x10C907: main (in /usr/bin/medusa)
==66586== 
==66586== 20 bytes in 1 blocks are definitely lost in loss record 5 of 19
==66586==    at 0x48489F3: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x485810E: initModule (in /usr/lib/x86_64-linux-gnu/medusa/modules/ssh.mod)
==66586==    by 0x4858314: go (in /usr/lib/x86_64-linux-gnu/medusa/modules/ssh.mod)
==66586==    by 0x10EAA0: startModule (in /usr/bin/medusa)
==66586==    by 0x110D9C: ??? (in /usr/bin/medusa)
==66586==    by 0x5063731: start_thread (pthread_create.c:447)
==66586==    by 0x50DE0FF: clone (clone.S:100)
==66586== 
==66586== 35 bytes in 1 blocks are still reachable in loss record 6 of 19
==66586==    at 0x4843808: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x4024C9A: malloc (rtld-malloc.h:56)
==66586==    by 0x4024C9A: strdup (strdup.c:42)
==66586==    by 0x4013294: _dl_load_cache_lookup (dl-cache.c:499)
==66586==    by 0x4007B5C: _dl_map_object (dl-load.c:2076)
==66586==    by 0x40027A0: openaux (dl-deps.c:64)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x4002B09: _dl_map_object_deps (dl-deps.c:232)
==66586==    by 0x400B3AC: dl_open_worker_begin (dl-open.c:638)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x400AB15: dl_open_worker (dl-open.c:803)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x400AEE7: _dl_open (dl-open.c:905)
==66586== 
==66586== 35 bytes in 1 blocks are still reachable in loss record 7 of 19
==66586==    at 0x4843808: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x400A890: malloc (rtld-malloc.h:56)
==66586==    by 0x400A890: _dl_new_object (dl-object.c:199)
==66586==    by 0x4005F5E: _dl_map_object_from_fd (dl-load.c:1042)
==66586==    by 0x4007930: _dl_map_object (dl-load.c:2209)
==66586==    by 0x40027A0: openaux (dl-deps.c:64)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x4002B09: _dl_map_object_deps (dl-deps.c:232)
==66586==    by 0x400B3AC: dl_open_worker_begin (dl-open.c:638)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x400AB15: dl_open_worker (dl-open.c:803)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x400AEE7: _dl_open (dl-open.c:905)
==66586== 
==66586== 56 bytes in 1 blocks are still reachable in loss record 8 of 19
==66586==    at 0x4843808: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x4BCA9A8: CRYPTO_malloc (in /usr/lib/x86_64-linux-gnu/libcrypto.so.3)
==66586==    by 0x4BCAB2C: CRYPTO_zalloc (in /usr/lib/x86_64-linux-gnu/libcrypto.so.3)
==66586==    by 0x4BDD940: CRYPTO_THREAD_lock_new (in /usr/lib/x86_64-linux-gnu/libcrypto.so.3)
==66586==    by 0x4AE131C: ??? (in /usr/lib/x86_64-linux-gnu/libcrypto.so.3)
==66586==    by 0x506856E: __pthread_once_slow (pthread_once.c:116)
==66586==    by 0x4BDDA3C: CRYPTO_THREAD_run_once (in /usr/lib/x86_64-linux-gnu/libcrypto.so.3)
==66586==    by 0x4AE150C: ??? (in /usr/lib/x86_64-linux-gnu/libcrypto.so.3)
==66586==    by 0x4AE06D2: CONF_modules_load (in /usr/lib/x86_64-linux-gnu/libcrypto.so.3)
==66586==    by 0x4AE0FD5: CONF_modules_load_file_ex (in /usr/lib/x86_64-linux-gnu/libcrypto.so.3)
==66586==    by 0x4AE12C7: ??? (in /usr/lib/x86_64-linux-gnu/libcrypto.so.3)
==66586==    by 0x4BC998E: ??? (in /usr/lib/x86_64-linux-gnu/libcrypto.so.3)
==66586== 
==66586== 59 bytes in 2 blocks are definitely lost in loss record 9 of 19
==66586==    at 0x48489F3: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x10DEB5: invokeModule (in /usr/bin/medusa)
==66586==    by 0x10EAA0: startModule (in /usr/bin/medusa)
==66586==    by 0x110D9C: ??? (in /usr/bin/medusa)
==66586==    by 0x5063731: start_thread (pthread_create.c:447)
==66586==    by 0x50DE0FF: clone (clone.S:100)
==66586== 
==66586== 64 bytes in 1 blocks are indirectly lost in loss record 10 of 19
==66586==    at 0x48489F3: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x10F652: loadLoginInfo (in /usr/bin/medusa)
==66586==    by 0x10C9CE: main (in /usr/bin/medusa)
==66586== 
==66586== 98 bytes in 2 blocks are still reachable in loss record 11 of 19
==66586==    at 0x4843808: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x4024C9A: malloc (rtld-malloc.h:56)
==66586==    by 0x4024C9A: strdup (strdup.c:42)
==66586==    by 0x40078A9: _dl_map_object (dl-load.c:2142)
==66586==    by 0x400B344: dl_open_worker_begin (dl-open.c:578)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x400AB15: dl_open_worker (dl-open.c:803)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x400AEE7: _dl_open (dl-open.c:905)
==66586==    by 0x505F7E7: dlopen_doit (dlopen.c:56)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x40015AE: _dl_catch_error (dl-catch.c:260)
==66586==    by 0x505F2F6: _dlerror_run (dlerror.c:138)
==66586== 
==66586== 98 bytes in 2 blocks are still reachable in loss record 12 of 19
==66586==    at 0x4843808: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x400A890: malloc (rtld-malloc.h:56)
==66586==    by 0x400A890: _dl_new_object (dl-object.c:199)
==66586==    by 0x4005F5E: _dl_map_object_from_fd (dl-load.c:1042)
==66586==    by 0x4007930: _dl_map_object (dl-load.c:2209)
==66586==    by 0x400B344: dl_open_worker_begin (dl-open.c:578)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x400AB15: dl_open_worker (dl-open.c:803)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x400AEE7: _dl_open (dl-open.c:905)
==66586==    by 0x505F7E7: dlopen_doit (dlopen.c:56)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x40015AE: _dl_catch_error (dl-catch.c:260)
==66586== 
==66586== 175 (88 direct, 87 indirect) bytes in 1 blocks are definitely lost in loss record 13 of 19
==66586==    at 0x48489F3: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x10F7F0: loadLoginInfo (in /usr/bin/medusa)
==66586==    by 0x10C9CE: main (in /usr/bin/medusa)
==66586== 
==66586== 472 bytes in 1 blocks are still reachable in loss record 14 of 19
==66586==    at 0x4843808: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x504F26A: __fopen_internal (iofopen.c:65)
==66586==    by 0x10EB38: loadFile (in /usr/bin/medusa)
==66586==    by 0x10CCC2: main (in /usr/bin/medusa)
==66586== 
==66586== 480 bytes in 3 blocks are still reachable in loss record 15 of 19
==66586==    at 0x48489F3: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x401274B: calloc (rtld-malloc.h:44)
==66586==    by 0x401274B: _dl_check_map_versions (dl-version.c:280)
==66586==    by 0x400B3F2: dl_open_worker_begin (dl-open.c:646)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x400AB15: dl_open_worker (dl-open.c:803)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x400AEE7: _dl_open (dl-open.c:905)
==66586==    by 0x505F7E7: dlopen_doit (dlopen.c:56)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x40015AE: _dl_catch_error (dl-catch.c:260)
==66586==    by 0x505F2F6: _dlerror_run (dlerror.c:138)
==66586==    by 0x505F898: dlopen_implementation (dlopen.c:71)
==66586==    by 0x505F898: dlopen@@GLIBC_2.34 (dlopen.c:81)
==66586== 
==66586== 1,176 bytes in 30 blocks are definitely lost in loss record 16 of 19
==66586==    at 0x4843808: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x66EE0FB: libssh2_session_last_error (in /usr/lib/x86_64-linux-gnu/libssh2.so.1.0.1)
==66586==    by 0x4857815: tryLogin (in /usr/lib/x86_64-linux-gnu/medusa/modules/ssh.mod)
==66586==    by 0x4857F10: initModule (in /usr/lib/x86_64-linux-gnu/medusa/modules/ssh.mod)
==66586==    by 0x4858314: go (in /usr/lib/x86_64-linux-gnu/medusa/modules/ssh.mod)
==66586==    by 0x10EAA0: startModule (in /usr/bin/medusa)
==66586==    by 0x110D9C: ??? (in /usr/bin/medusa)
==66586==    by 0x5063731: start_thread (pthread_create.c:447)
==66586==    by 0x50DE0FF: clone (clone.S:100)
==66586== 
==66586== 1,253 bytes in 1 blocks are still reachable in loss record 17 of 19
==66586==    at 0x48489F3: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x400A5B9: calloc (rtld-malloc.h:44)
==66586==    by 0x400A5B9: _dl_new_object (dl-object.c:92)
==66586==    by 0x4005F5E: _dl_map_object_from_fd (dl-load.c:1042)
==66586==    by 0x4007930: _dl_map_object (dl-load.c:2209)
==66586==    by 0x40027A0: openaux (dl-deps.c:64)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x4002B09: _dl_map_object_deps (dl-deps.c:232)
==66586==    by 0x400B3AC: dl_open_worker_begin (dl-open.c:638)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x400AB15: dl_open_worker (dl-open.c:803)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x400AEE7: _dl_open (dl-open.c:905)
==66586== 
==66586== 1,890 bytes in 30 blocks are definitely lost in loss record 18 of 19
==66586==    at 0x4843808: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x66F584F: ??? (in /usr/lib/x86_64-linux-gnu/libssh2.so.1.0.1)
==66586==    by 0x66E8E35: ??? (in /usr/lib/x86_64-linux-gnu/libssh2.so.1.0.1)
==66586==    by 0x66F762D: libssh2_userauth_list (in /usr/lib/x86_64-linux-gnu/libssh2.so.1.0.1)
==66586==    by 0x4857705: tryLogin (in /usr/lib/x86_64-linux-gnu/medusa/modules/ssh.mod)
==66586==    by 0x4857F10: initModule (in /usr/lib/x86_64-linux-gnu/medusa/modules/ssh.mod)
==66586==    by 0x4858314: go (in /usr/lib/x86_64-linux-gnu/medusa/modules/ssh.mod)
==66586==    by 0x10EAA0: startModule (in /usr/bin/medusa)
==66586==    by 0x110D9C: ??? (in /usr/bin/medusa)
==66586==    by 0x5063731: start_thread (pthread_create.c:447)
==66586==    by 0x50DE0FF: clone (clone.S:100)
==66586== 
==66586== 2,578 bytes in 2 blocks are still reachable in loss record 19 of 19
==66586==    at 0x48489F3: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==66586==    by 0x400A5B9: calloc (rtld-malloc.h:44)
==66586==    by 0x400A5B9: _dl_new_object (dl-object.c:92)
==66586==    by 0x4005F5E: _dl_map_object_from_fd (dl-load.c:1042)
==66586==    by 0x4007930: _dl_map_object (dl-load.c:2209)
==66586==    by 0x400B344: dl_open_worker_begin (dl-open.c:578)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x400AB15: dl_open_worker (dl-open.c:803)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x400AEE7: _dl_open (dl-open.c:905)
==66586==    by 0x505F7E7: dlopen_doit (dlopen.c:56)
==66586==    by 0x4001488: _dl_catch_exception (dl-catch.c:241)
==66586==    by 0x40015AE: _dl_catch_error (dl-catch.c:260)
==66586== 
==66586== LEAK SUMMARY:
==66586==    definitely lost: 3,256 bytes in 66 blocks
==66586==    indirectly lost: 87 bytes in 3 blocks
==66586==      possibly lost: 0 bytes in 0 blocks
==66586==    still reachable: 5,105 bytes in 14 blocks
==66586==         suppressed: 0 bytes in 0 blocks
==66586== 
==66586== ERROR SUMMARY: 7 errors from 7 contexts (suppressed: 0 from 0)
